### PR TITLE
chore: add logging to compaction

### PR DESCRIPTION
**Pull request by davidby-influx**
Closes https://github.com/influxdata/influxdb/issues/21704

Add progress logging and more detailed error logging for compaction and compaction-related procedures.

- [X] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [X] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [X] Rebased/mergeable
- [X] Tests pass

### CODE REVIEW

File: tsdb/engine/tsm1/compact.go [hunk 1]
```diff
@@ -1061,16 +1073,15 @@ func (c *Compactor) writeNewFiles(generation, sequence int, src []string, iter K
 			// planner keeps track of which files are assigned to compaction plans now.
 			return nil, err
 		} else if err != nil {
+			// We hit an error and didn't finish the compaction.  Abort.
 			// Remove any tmp files we already completed
+			// discard later errors to return the first one from the write() call
 			for _, f := range files {
-				if err := os.RemoveAll(f); err != nil {
```
Comment (codyshepherd) [inline tsdb/engine/tsm1/compact.go:1066]: Why the change in error semantics, and does this risk breaking error handling expectations elsewhere, e.g. in compactionStrategy.compactGroup()? I don't have enough clarity on what kinds of errors, in what order, would be returned in the unroll by the old code.

**Review by codyshepherd — APPROVED** [review]
A couple minor notes and a question about the change to error handling/returning in Compactor.writeNewFiles

File: tsdb/engine/tsm1/compact.go [hunk 2]
```diff
@@ -211,9 +213,15 @@ func (c *DefaultPlanner) ParseFileName(path string) (int, int, error) {
 }
 
 // FullyCompacted returns true if the shard is fully compacted.
```
[the remark this answers is not in the document]
Comment (davidby-influx) [reply tsdb/engine/tsm1/compact.go:215]: Will do.

File: tsdb/engine/tsm1/compact.go (hunk 1 above)
Comment (davidby-influx) [reply tsdb/engine/tsm1/compact.go:1066]: We hit an error from the c.write() call. That's the most important error to log, because the first error has the most information about the initial problem. Overwriting it with a subsequent error from a clean-up call can mask the root of the problem. We don't care what errors the RemoveAll() calls return, because we failed before them.

File: tsdb/engine/tsm1/engine.go [hunk 3]
```diff
@@ -916,8 +916,8 @@ func (e *Engine) IsIdle() (state bool, reason string) {
 
 	if cacheSize := e.Cache.Size(); cacheSize > 0 {
 		return false, "not idle because cache size is nonzero"
-	} else if !e.CompactionPlan.FullyCompacted() {
-		return false, "not idle because shard is not fully compacted"
+	} else if c, r := e.CompactionPlan.FullyCompacted(); !c {
```
[the remark this answers is not in the document]
Comment (davidby-influx) [reply tsdb/engine/tsm1/engine.go:919]: I prefer the clarity of `false` - saves the user having to scan back up to verify that `c` was the test condition.  Just a style issue, either is fine.
